### PR TITLE
fix(web-components): add ability to hide close icon and fix spacing around buttons

### PR DESCRIPTION
**Pull request by dn55533**
## Summary of the changes

1. Add a new hideCloseButton prop to allow the ic-dialog close button to be hidden. This is an option in the Figma design kit but was missing from the library. Having the close button is the default if the prop is not provided.
2. Equalise the bottom and right hand margins around the ic-dialog call to action buttons (previously the bottom margin was smaller than the right hand margin).

## Related issue
N/A

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

**Comment by dz333dz** [comment]
NB - this is a duplicate of closed PR https://github.com/mi6/ic-ui-kit/pull/1195. Re-raised in this PR due to merge/rebase issues.

**Review by GCHQ-Developer-299 — APPROVED** [review]
cloned the repo and checked out the changes in storybook locally, it all functions fine :)

**Review by GCHQ-Developer-299 — CHANGES_REQUESTED** [review]
Sorry to revoke my approval - because you've added a new prop to dialog, we'd expect to see an update to docs.json as well (it's automatically generated when you build)

**Review by DBD324 — CHANGES_REQUESTED** [review]
I think that the ability to show and hide the close icon is valuable in some use cases, but we should make sure the default is for this to be displayed. 

With regards to the padding changes, i agree the bottom padding should be increased. I'm not sure however that the same should be said for the top padding. The designs purposefully only specify 8px padding at the top to account for the visual whitespace in the icon button area when displaying its default state. We should aim for 8px top padding, with 16px on the sides and bottom for now to account for this optical alignment issue.

**Comment by dz333dz** [comment]
> I think that the ability to show and hide the close icon is valuable in some use cases, but we should make sure the default is for this to be displayed.
> 
> With regards to the padding changes, i agree the bottom padding should be increased. I'm not sure however that the same should be said for the top padding. The designs purposefully only specify 8px padding at the top to account for the visual whitespace in the icon button area when displaying its default state. We should aim for 8px top padding, with 16px on the sides and bottom for now to account for this optical alignment issue.

The default for the close button is to be displayed if the hideCloseButton prop is not provided, so there will be no change in behaviour for users of the component.

Happy to update the CSS so that only the bottom margin is increased, not the top one too.

**Comment by dz333dz** [comment]
> Sorry to revoke my approval - because you've added a new prop to dialog, we'd expect to see an update to docs.json as well (it's automatically generated when you build)

Will re-run the build and include the updated docs.json file in this commit.

**Comment by dz333dz** [comment]
All PR comments implemented, ready for re-review.

**Review by GCHQ-Developer-299 — APPROVED** [review]
Thank you for making those changes so promptly :)

File: packages/web-components/src/components/ic-dialog/ic-dialog.css
Comment (GCHQ-Developer-299) [inline packages/web-components/src/components/ic-dialog/ic-dialog.css:1]: I've run the build Checks and apparently there's a Prettier linting error in this file now.

**Review by GCHQ-Developer-299 — CHANGES_REQUESTED** [review]

File: packages/web-components/src/components/ic-dialog/ic-dialog.css
Comment (dz333dz) [reply packages/web-components/src/components/ic-dialog/ic-dialog.css:1]: Fixed prettier problem.

**Review by GCHQ-Developer-847 — APPROVED** [review]
Looks great to me - thanks again for your contribution and responding to all our comments so quickly!

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by DBD324 — APPROVED** [review]

**Pull request merged**